File: ingester/internal/events/transfer_event.go
package events

// TransferEvent represents LP token movement for the pair contract token.
//
// This is used to correlate Mint/Burn activity with LP token accounting.
type TransferEvent struct {
	BaseEvent
	From  string `json:"from"`
	To    string `json:"to"`
	Value string `json:"value"`
}

func (e TransferEvent) ToMap() map[string]interface{} {
	m := e.BaseEvent.ToMap()
	m["from"] = e.From
	m["to"] = e.To
	m["value"] = e.Value
	return m
}

File: aggregator/src/main/java/com/web3analytics/models/TransferEvent.java
package com.web3analytics.models;

/**
 * Transfer event from dex-liquidity-events topic.
 * Represents LP token movement on the pair contract token.
 */
public record TransferEvent(
        String eventId,
        long blockNumber,
        long blockTimestamp,
        String transactionHash,
        int logIndex,
        String pairAddress,
        String token0,
        String token1,
        String token0Symbol,
        String token1Symbol,
        String from,
        String to,
        String value,
        long eventTimestamp
) implements DexEvent {}
